ivy/stateful/module.py

```diff
@@ -817,6 +817,16 @@ def register_buffer(self, var_name, value):
         """Set the buffer at any place within the class."""
         self._set_buffers({var_name: value})
 
+    def eval(self, mode: bool = False):
```

Comment:
Actually `eval` doesn't need to receive a `mode` argument, basically `model.train()` would set `training` to `True` and `model.eval()` would set `training` to `False` as used in `torch.nn.Module`. Also `self.training` should be `True` by default, so we should be setting that in the `__init__` itself

Reply:
There's a minor bug with this logic where if a user passes `model.eval(True)`, it would set `training=True`